ratom/views.py

```diff
@@ -1,3 +1,178 @@
-from django.shortcuts import render
+from rest_framework import status
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
 
-# Create your views here.
+from django_elasticsearch_dsl_drf import constants, filter_backends
+from django_elasticsearch_dsl_drf.viewsets import DocumentViewSet
+from elasticsearch_dsl import DateHistogramFacet, RangeFacet, TermsFacet
+from ratom.permissions import IsOwnerOrReadOnly
+
+from .documents.message import MessageDocument
+from .models import User, Account, Message
+from .serializers import (
+    UserSerializer,
+    AccountSerializer,
+    MessageSerializer,
+    MessageDocumentSerializer,
+)
+
+
+@api_view(["GET"])
+@permission_classes([IsAuthenticated])
+def user_detail(request):
+    """
+    Show details of single user
+    """
+    try:
+        user_pk = request.user.pk
+        user = User.objects.get(pk=user_pk)
+    except User.DoesNotExist:
+        return Response(status=status.HTTP_404_NOT_FOUND)
+
+    if request.method == "GET":
+        serialized_user = UserSerializer(user)
+        return Response(serialized_user.data)
+
+
+@api_view(["GET", "POST"])
+@permission_classes([IsAuthenticated])
+def account_list(request):
+    """
+    List all Accounts, or create a new Account.
+    """
+    if request.method == "GET":
+        # TODO: currently showing all accounts.
+        # TODO: need to decide how/if to limit access
+        accounts = Account.objects.all()
+        serialized_account = AccountSerializer(accounts, many=True)
+        return Response(serialized_account.data)
+
+    if request.method == "POST":
+        serialized_account = AccountSerializer(data=request.data)
+        if serialized_account.is_valid():
+            serialized_account.save()
+            return Response(serialized_account.data, status=status.HTTP_201_CREATED)
```

Comment:
Now is probably a good time to agree on what we should be unit testing. Ideally all of these views would include unit tests. Happy to help write some up.